### ISSUE TITLE
Exception when returning a stackonly structure in a benchmark case using in-process toolchain

### ISSUE THREAD

**Issue by alexanderkozlenko** [dotnet/BenchmarkDotNet#843]
A `System.Reflection.TargetInvocationException` exception is occurring when returning a stack-only value type in a benchmark case using in-process toolchain:

```
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.TypeLoadException: The generic type 'BenchmarkAction`1' was used with an invalid instantiation in assembly 'BenchmarkDotNet, Version=0.11.0.0, Culture=neutral, PublicKeyToken=aa0ca2f9092cefc4'.
   at System.RuntimeType.MakeGenericType(Type[] instantiation)
   at BenchmarkDotNet.Toolchains.InProcess.BenchmarkActionFactory.CreateCore(Object instance, MethodInfo targetMethod, MethodInfo fallbackIdleSignature, BenchmarkActionCodegen codegenMode, Int32 unrollFactor)
   at BenchmarkDotNet.Toolchains.InProcess.InProcessRunner.Runnable.RunCore(IHost host, BenchmarkCase benchmarkCase, BenchmarkActionCodegen codegenMode, IConfig config)
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at BenchmarkDotNet.Toolchains.InProcess.InProcessRunner.Run(IHost host, BenchmarkCase benchmarkCase, BenchmarkActionCodegen codegenMode, IConfig config)
```

- BenchmarkDotNet: 0.11.0
- Target Framework: `netcoreapp2.1` (2.1.2.26629)
- Log: [ConsoleApp1.log](https://github.com/dotnet/BenchmarkDotNet/files/2227072/ConsoleApp1.log)
- Project: [ConsoleApp1.zip](https://github.com/dotnet/BenchmarkDotNet/files/2227075/ConsoleApp1.zip)

**Comment by AndreyAkinshin** [comment]
@alexanderkozlenko, thanks for the repro!
@adamsitnik, could you take a look?
```cs
public class Program
{
    public static void Main() => BenchmarkRunner.Run<BenchmarkType>();
}

[InProcess]
public class BenchmarkType
{
    [Benchmark]
    public TestType BenchmarkCase() => default;
}

public ref struct TestType
{
}
```

**Comment by adamsitnik** [comment]
Hi @alexanderkozlenko 

Unfortunately, this feature was not ported to `InProcessToolchain` yet. It was easy to change the C# code we generate, but updating the `InProcessToolchain` which needs to generate the right IL is non-trivial.

I don't expect to implement it personally any time soon (I am too busy with porting CoreFX to BenchmarkDotNet).

**Comment by adamsitnik** [comment]
Fixed by #921.

**Comment by adamsitnik** [comment]
@alexanderkozlenko please keep in mind that we kept the old InProcessToolchain and added a new one `InProcessEmitToolchain`. The new one has the fix, the old one does not
